File: programming-challenges/mc821/10-18/password.cpp
// TODO gets TLE

#include <bits/stdc++.h>
using namespace std;

int main(){
    string s;
    getline(cin, s);

    string ans("");
    for(int i = s.size(); i > 0 ; i--){
        string prefix = s.substr(0, i);
        string suffix = s.substr(s.size()-i);
        if(prefix == suffix){
            int mid_i = s.find(prefix, 1);
            if(mid_i == string::npos || mid_i >= (s.size()-i))
                continue;
            ans = prefix;
            break;
        }
    }
    std::cout << ((ans.size() > 0) ? ans : "Just a legend") << std::endl;

    return 0;
}

File: programming-challenges/mc821/10-18/password.py
# TODO gets TlE

s = input()

for i in range(len(s),-1,-1):
    prefix = s[:i]
    sufix = s[-i:]
    if prefix == sufix:
        if prefix in s[1:-1]:
            print(prefix)
            break
    if i == 0:
        print("Just a legend")
